vmm/src/vm.rs

```diff
@@ -1048,13 +1088,19 @@ impl Vm {
     fn load_payload_async(
         memory_manager: &Arc<Mutex<MemoryManager>>,
         config: &Arc<Mutex<VmConfig>>,
+        #[cfg(feature = "igvm")] cpu_manager: &Arc<Mutex<cpu::CpuManager>>,
     ) -> Result<Option<thread::JoinHandle<Result<EntryPoint>>>> {
         // Kernel with TDX is loaded in a different manner
         #[cfg(feature = "tdx")]
         if config.lock().unwrap().is_tdx_enabled() {
             return Ok(None);
         }
 
+        #[cfg(feature = "igvm")]
+        if config.lock().unwrap().is_igvm_provided() {
+            return Ok(None);
+        }
+
```

Comment:
Why you want to skip for igvm? If this is a workaround for #5993, it should be done from `load_igvm()` where you have the comments about the related issue.

Reply:
I have followed the TDX one above here. It's just dealy load without loader_handle.

Reply:
This is not needed anymore.